packages/core/src/worker/Source.worker.ts

```diff
@@ -5,14 +5,22 @@ import fs from 'fs';
 import { DirectoryJSON, Volume } from 'memfs';
 import { switchMap, tap } from 'rxjs';
 
-import type { Page, WorkerData } from '@jpmorganchase/mosaic-types';
+import type { Page, PluginErrors, WorkerData } from '@jpmorganchase/mosaic-types';
 
 import FileAccess from '../filesystems/FileAccess.js';
 import MutableVolume from '../filesystems/MutableVolume.js';
 import createConfig from '../helpers/createConfig.js';
 import { bindPluginMethods, bindSerialiser } from '../plugin/index.js';
 import createSourceObservable from './helpers/createSourceObservable.js';
 
+const trackPluginErrors = (errors: PluginErrors, lifecycleMethod: string) => {
+  const errorsBuffer = Buffer.from(JSON.stringify({ errors, lifecycleMethod }));
+  parentPort.postMessage({
```

Comment:
Plugin methods that start with a `$` run inside a child process and must communicate with the parent thread by posting a message.